rider-fsharp/protocol/src/kotlin/model/RdFSharpTypeProvidersLoaderModel.kt

```diff
@@ -0,0 +1,157 @@
+    }
+
+    private val ProvidedNamespace = structdef {
+        call("getNestedNamespaces", void, immutableList(string))
+        call("getTypes", void, immutableList(Type))
+        call("resolveTypeName", string, Type)
+
+        field("namespaceName", string)
+    }
+/*
+    private val TypeProvider = structdef"TypeProvider") {
+        sink("invalidate", void).async
+
+        call("getNamespaces", void, immutableList(string))
+        call("applyStaticArguments", Type, immutableList(ParameterInfo))
+        call("getInvokerExpression", int, int)
+        call("getGeneratedAssemblyContents", void, immutableList(byte))
+    }*/
+
+    private val ApplyStaticArgumentsParameters = structdef {
+        field("typeProviderId", int)
+        field("typeWithoutArguments", string)
+        field("typePathWithArguments", array(string))
+        field("staticArguments", array(string))
+    }
+
+    private val GetStaticArgumentsParameters = structdef {
+        field("typeProviderId", int)
+        field("typeWithoutArguments", string)
+    }
+
+    private val GetInvokerExpressionParameters = structdef {
+        field("typeProviderId", int)
+        field("syntheticMethodBase", string)
+        field("parameters", string)
+    }
+
+    private val GetGeneratedAssemblyContentsParameters = structdef {
+        field("typeProviderId", int)
+        field("syntheticMethodBase", string)
+        field("parameters", string)
+    }
+
+    private val pos = structdef {
+        field("Line", int)
+        field("Column", int)
+        field("Encoding", int)
+        field("EncodingSize", int)
+    }
+
+    private val range1 = structdef {
+        field("StartLine", int)
+        field("StartColumn", int)
+        field("EndLine", int)
+        field("EndColumn", int)
+        field("Start", pos)
+        field("End", pos)
+        field("StartRange", this)
+        field("EndRange", this)
+        field("FileIndex", int)
+        field("FileName", string)
+        field("IsSynthetic", bool)
+    }
+
+    private val PublicKey = structdef {
+        field("IsKey", bool)
+        field("IsKeyToken", bool)
+        field("Key", array(byte))
+        field("KeyToken", array(byte))
+    }
+
+    private val Version = structdef {
+        field("Major", int)
+        field("Minor", int)
+        field("Build", int)
+        field("Revision", int)
+    }
+
+    private val ILAssemblyRef = structdef {
+        field("Name", string)
+        field("QualifiedName", string)
+        field("Hash", array(byte))
+        field("PublicKey", PublicKey)
+        field("Retargetable", bool)
+        field("Version", Version)
+        field("Locale", string)
+    }
+
+    private val ILModuleRef = structdef {
+        field("Name", string)
+        field("HasMetadata", bool)
+        field("Hash", array(byte))
+    }
+
+    private val ILScopeRef = structdef {
+        field("IsModuleRef", bool)
+        field("IsAssemblyRef", bool)

```

Comment:
Maybe use enum instead?

Reply:
> Please use enum instead.

corrected for models in which the total size of boolean fields is more than 4 bytes (sizeof(int/enum))